sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/InferSchema.scala

```diff
@@ -253,7 +253,7 @@ private[sql] object InferSchema {
    * Returns the most general data type for two given data types.
    */
   def compatibleType(t1: DataType, t2: DataType): DataType = {
-    TypeCoercion.findTightestCommonTypeOfTwo(t1, t2).getOrElse {
+    TypeCoercion.findTightestCommonType(t1, t2).getOrElse {
```

Comment:
Oh, yea, it appearntly possible for now. But, can I maybe do this in another PR if you think it is fine? It seems possible but I would like to avoid to fix it here simply because of a worry to resolve a conflict in the PR refacorting this code path and it seems not directly related to the JIRA.